src/lockfile/wrapper.js

```diff
@@ -86,9 +87,12 @@ export default class Lockfile {
     [key: string]: LockManifest
   };
 
-  static async fromDirectory(dir: string, reporter?: Reporter): Promise<Lockfile> {
+  static async fromDirectory(config: Config, dir: string, reporter?: Reporter): Promise<Lockfile> {
```

Comment:
Yep, it should be fairly easy. I think most tools currently stored inside the `config` object (such as `tryManifest`, etc) could be safely moved somewhere else (for example in miniyarn, I had a `util/yarn` file to store all such functions), so that the config would only contain the configuration data and nothing else.